File: articles/sensitivity-analysis-for-algorithms-and-models/typescript/sensitivity_analysis.ts
function clamp(x: number, lo = 0, hi = 1): number {
  return Math.max(lo, Math.min(hi, x));
}
function model(demand: number, capacity: number, failure: number, adaptation: number): number {
  return clamp(0.5 + 0.30 * demand + 0.25 * failure - 0.20 * capacity - 0.15 * adaptation);
}
console.log(`baseline_risk=${model(0.45, 0.35, 0.25, 0.30).toFixed(6)}`);

File: articles/sensitivity-analysis-for-algorithms-and-models/java/SensitivityAnalysis.java
public class SensitivityAnalysis {
  static double clamp(double x) { return Math.max(0.0, Math.min(1.0, x)); }
  static double model(double demand, double capacity, double failure, double adaptation) {
    return clamp(0.5 + 0.30*demand + 0.25*failure - 0.20*capacity - 0.15*adaptation);
  }
  public static void main(String[] args) {
    System.out.printf("baseline_risk=%.6f%n", model(0.45, 0.35, 0.25, 0.30));
  }
}
